src/lender/deployer.sol

```diff
@@ -0,0 +1,243 @@
+// Copyright (C) 2020 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity >=0.5.12;
+import { Auth } from "tinlake-auth/auth.sol";
+
+// lender contracts
+import { DefaultAssessor } from "./assessor/default.sol";
+import { FullInvestmentAssessor } from "./assessor/full_investment.sol";
+import { AllowanceOperator } from "./tranche/operator/allowance.sol";
+import { WhitelistOperator } from "./tranche/operator/whitelist.sol";
+import { Tranche } from "./tranche/tranche.sol";
+import { SeniorTranche } from "./tranche/senior_tranche.sol";
+import { DefaultDistributor } from "./distributor/default.sol";
+
+import "tinlake-erc20/erc20.sol";
+
+contract AuthLike {
+    function rely(address) public;
+    function deny(address) public;
+}
+
+contract DependLike {
+    function depend(bytes32, address) public;
+}
+
+
+
+contract TrancheFab {
+    string constant public symbol = "TIN";
+    string constant public name = "Tinlake TIN Token";
+
+    function newTranche(address currency, address assessor) public returns (address) {
+        ERC20 token = new ERC20(symbol, name);
+        Tranche tranche = new Tranche(address(token), currency);
+        tranche.rely(msg.sender);
+        tranche.deny(address(this));
+        token.rely(address(tranche));
+        return address(tranche);
+    }
+}
+
+contract SeniorTrancheFab {
+    string constant public symbol = "DROP";
+    string constant public name = "Tinlake Drop Token";
+    uint public ratePerSecond;
+
+    constructor (uint rate_) public {
+        ratePerSecond = rate_;
+    }
```

Comment:
We want to re-use the Fabs on mainnet for different deployments. (with different rates)

I think each deployment will have a different token name

It should not be required to touch a solidity file if we want to deploy for another asset group.

Reply:
I agree, but I didn't want to blow up this even more. I think the added security for now is better than the flexibility we had. I will try to add it in a separate PR.